### PR TITLE
mobile: Switch to Truth from AssertJ

### PR DESCRIPTION
**Pull request by fredyw**
Having multiple assertion libraries can be confusing. This PR switches the Java and Kotlin tests to use Google's Truth instead of AssertJ since that is what's used at Google, which makes rebasing Cronet for example easier. Google's Truth has also better failure messages. For comparison between the two, see [this](https://truth.dev/comparison.html).

Risk Level: low (tests only)
Testing: CI
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt
Comment (RyanTheOptimist) [inline mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt:1]: This isn't really relevant to your PR, but should we consider moving these mocks out of library/kotlin and into test/kotlin? (assuming they're test-only?)

**Review by RyanTheOptimist — APPROVED** [review]
Sweet! Thanks for doing this!

**Review by abeyad — APPROVED** [review]

File: mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt
Comment (fredyw) [reply mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt:1]: Yeah, good idea. I can do that in a separate PR.

Comment (abeyad) [reply mobile/library/kotlin/io/envoyproxy/envoymobile/mocks/MockEnvoyHTTPStream.kt:1]: I think the whole mocks directory should go to test/kotlin

**Pull request merged**